client/fvESg_service.go

```diff
@@ -0,0 +1,389 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/ciscoecosystem/aci-go-client/container"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func (sm *ServiceManager) CreateEndpointSecurityGroup(name string ,application_profile string ,tenant string , description string, fvESgattr models.EndpointSecurityGroupAttributes) (*models.EndpointSecurityGroup, error) {	
+	rn := fmt.Sprintf(models.Rn,name)
+	parentDn := fmt.Sprintf(models.ParentDn, tenant ,application_profile )
+	fvESg := models.NewEndpointSecurityGroup(rn, parentDn, description, fvESgattr)
+	err := sm.Save(fvESg)
+	return fvESg, err
+}
+
+func (sm *ServiceManager) ReadEndpointSecurityGroup(name string ,application_profile string ,tenant string ) (*models.EndpointSecurityGroup, error) {
+	dn := fmt.Sprintf(models.Dn, tenant ,application_profile ,name )    
+	cont, err := sm.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	fvESg := models.EndpointSecurityGroupFromContainer(cont)
+	return fvESg, nil
+}
+
+func (sm *ServiceManager) DeleteEndpointSecurityGroup(name string ,application_profile string ,tenant string ) error {
+	dn := fmt.Sprintf(models.Dn, tenant ,application_profile ,name )
+	return sm.DeleteByDn(dn, models.FvesgClassName)
+}
+
+func (sm *ServiceManager) UpdateEndpointSecurityGroup(name string ,application_profile string ,tenant string  ,description string, fvESgattr models.EndpointSecurityGroupAttributes) (*models.EndpointSecurityGroup, error) {
+	rn := fmt.Sprintf(models.Rn,name)
+	parentDn := fmt.Sprintf(models.ParentDn, tenant ,application_profile )
+	fvESg := models.NewEndpointSecurityGroup(rn, parentDn, description, fvESgattr)
+    fvESg.Status = "modified"
+	err := sm.Save(fvESg)
+	return fvESg, err
+}
+
+func (sm *ServiceManager) ListEndpointSecurityGroup(application_profile string ,tenant string ) ([]*models.EndpointSecurityGroup, error) {
+	dnUrl := fmt.Sprintf("%s/models.ParentDn/fvESg.json", models.BaseurlStr , tenant ,application_profile )
+    cont, err := sm.GetViaURL(dnUrl)
+	list := models.EndpointSecurityGroupListFromContainer(cont)
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsSecInheritedFromEndpointSecurityGroup( parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rssecInherited-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.